### PR TITLE
CNV-26881: merge DataSource and Bootable volumes part 1

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Ticket: https://issues.redhat.com/browse/CNV-26881
Design: https://docs.google.com/document/d/1oETHC4UpRnEGFEdiezUxMLzpQagWXOYz0DBYro-XHB0/edit#

What i did in this pr:
- removed DataSource navbar
- added pvc to the Bootable volumes list

What i have to do in the following prs:
- change 'Bootable volumes' into 'Bootable resources'
- change the creation modal to accommodate the registry method.
- change the modal volume metadata section according to design
- change actions according to design

## 🎥 Demo

image

### CODE REVIEW

File: src/views/bootablevolumes/list/components/BootableVolumesRow.tsx [hunk 1]
```diff
@@ -1,49 +1,66 @@
...
+              groupVersionKind={
+                obj.kind === DataSourceModelGroupVersionKind.kind
+                  ? DataSourceModelGroupVersionKind
+                  : PersistentVolumeClaimModelGroupVersionKind
+              }
+              name={obj?.metadata?.name}
+            />
           </SplitItem>
-          {isDataSourceCloning(obj) && (
+          {obj.kind === DataSourceModel.kind && isDataSourceCloning(obj as V1beta1DataSource) && (
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/views/bootablevolumes/list/components/BootableVolumesRow.tsx:54]: Typescript throws an error as the method accept just data sources. But in future we can create a method that support also pvcs

File: src/views/bootablevolumes/list/components/BootableVolumesRow.tsx [hunk 2]
```diff
@@ -1,49 +1,66 @@
...
   return (
     <>
-      <TableData id="name" activeColumnIDs={activeColumnIDs} className="pf-m-width-15">
+      <TableData id="name" activeColumnIDs={activeColumnIDs} className="pf-m-width-20">
         <Split hasGutter>
           <SplitItem>
-            <ResourceLink groupVersionKind={groupVersionKind} name={obj?.metadata?.name} />
+            <ResourceLink
+              groupVersionKind={
+                obj.kind === DataSourceModelGroupVersionKind.kind
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/views/bootablevolumes/list/components/BootableVolumesRow.tsx:47]: Thanks

File: src/views/bootablevolumes/utils/types.ts [hunk 3]
```diff
@@ -1,4 +1,9 @@
+import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+
 export type BootableVolumeMetadata = {
   labels: { [key: string]: string };
   annotations: { [key: string]: string };
 };
+
+export type BootableResource = V1beta1DataSource | IoK8sApiCoreV1PersistentVolumeClaim;
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/views/bootablevolumes/utils/types.ts:9]: Maybe in a separate pr i can make that shared with the modal that is used in the bootablevolumes and catalog page.
WDYT?

File: src/views/bootablevolumes/list/BootableVolumesList.tsx [hunk 4]
```diff
@@ -47,6 +46,8 @@ const BootableVolumesList: FC = () => {
     }));
   };
 
+  console.log(data);
```
[the remark this answers is not in the document]
Comment (avivtur) [reply src/views/bootablevolumes/list/BootableVolumesList.tsx:49]: ?

File: src/views/bootablevolumes/list/components/BootableVolumesRow.tsx [hunk 5]
```diff
@@ -1,49 +1,67 @@
...
   return (
     <>
-      <TableData id="name" activeColumnIDs={activeColumnIDs} className="pf-m-width-15">
+      <TableData id="name" activeColumnIDs={activeColumnIDs} className="pf-m-width-20">
         <Split hasGutter>
           <SplitItem>
-            <ResourceLink groupVersionKind={groupVersionKind} name={obj?.metadata?.name} />
+            <ResourceLink
+              groupVersionKind={
+                isBootableVolumePVCKind(obj)
```
Comment (avivtur) [inline src/views/bootablevolumes/list/components/BootableVolumesRow.tsx:48]: I just remembered you can use this:
https://github.com/kubevirt-ui/kubevirt-plugin/blob/318c9c2180e53edea29a14e9d86b13b7429981a4/src/views/catalog/CreateFromInstanceTypes/utils/utils.ts#L144-L147

Comment (upalatucci) [reply src/views/bootablevolumes/list/components/BootableVolumesRow.tsx:48]: Great :-)

**Pull request merged**